src/opm/output/eclipse/Tables.cpp

```diff
@@ -41,8 +48,100 @@
+    /// keyword.
+    ///
+    /// \tparam BuildDependent Callable entity that extracts the
+    ///    independent and primary dependent variates of a single
+    ///    saturation function table into a linearised output table.
+    ///    Must implement a function call operator of the form
+    ///    \code
+    ///       std::size_t
+    ///       operator()(const std::size_t      tableID,
+    ///                  const std::size_t      primID,
+    ///                  LinearisedOutputTable& table);
+    ///    \endcode
+    ///    that will assign the independent variate of the sub-table
+    ///    primID within the table identified as tableID to column zero
+    ///    of 'table' and all dependent variates to columns one &c of
+    ///    'table'.  The function call operator must return the number
+    ///    of active (used) rows within the sub-table through its return
+    ///    value.
+    ///
+    /// \param[in] numTab Number of tables in this table collection.
+    ///
+    /// \param[in] numPrim Number of primary look-up keys for each table.
+    ///    Mostly relevant (i.e., greater than one) for miscible oil
+    ///    ("PVTO") or miscible gas ("PVTG") tables and typically
+    ///    corresponds to the number of Rs/Rv entries from the TABDIMS
+    ///    keyword.
+    ///
+    /// \param[in] numRows Number of rows to allocate for each padded
+    ///    output table.
+    ///
+    /// \param[in] numDep Number of dependent variates (columns) in each
+    ///    table of this collection of input tables.  Total number of
+    ///    columns in the result vector will be 1 + 2*numDep to account
+    ///    for the independent variate, the dependent variates and the
+    ///    derivatives of the dependent variates with respect to the
+    ///    independent variate.
+    ///
+    /// \param[in] buildDeps Function object that implements the
+    ///    protocol outlined for \code BuildDependent::operator()()
+    ///    \endcode.  Typically a lambda expression.
+    ///
+    /// \return Linearised, padded TAB vector entries for a collection of
+    ///    tabulated property functions (e.g., saturation functions or PVT
+    ///    functions) corresponding to a single input keyword.  Derivatives
+    ///    included as additional columns.
+    template <class BuildDependent>
+    std::vector<double>
+    createPropfuncTable(const std::size_t numTab,
+                        const std::size_t numPrim,
+                        const std::size_t numRows,
+                        const std::size_t numDep,
+                        const double      fillVal,
+                        BuildDependent&&  buildDeps)
+    {
+        const auto numCols = 1 + 2*numDep;
+
+        auto descr = ::Opm::DifferentiateOutputTable::Descriptor{};
+        descr.primID = 0 * numPrim;
+
+        auto linTable = ::Opm::LinearisedOutputTable {
+            numTab, numPrim, numRows, numCols, fillVal
+        };
+
+        for (descr.tableID = 0*numTab;
```

Comment:
Mutating the member in this way looked slightly funny - when did a normal index variable go out of fashion?

Reply:
> when did a normal index variable go out of fashion?

It probably didn't.  Here the mutation serves two purposes.  The first is obviously just controlling the loop.  The second is that the entire `descr` structure feeds into `calcSlopes()` to inform that operation of what subsection of the table it is supposed to analyse and for which to compute derivatives.  If we use the more traditional style of keeping separate indexing variables then we would need to construct a new table descriptor for each call to `calcSlopes()`.

Would you like me to add a comment to this effect?

Reply:
> Would you like me to add a comment to this effect?

yes - that'd be good.